rfc/0009-audit-checks.md

```diff
@@ -35,67 +35,50 @@ As a Kubernetes developer, I want access to the audit checks results to allow me
+policies in monitor mode. Failed evaluations for policies in enforce mode will not be stored as resources won't be 
+created in the cluster. These evaluations will be stored in a `PolicyReport`. There is one `PolicyReport` per namespace and 
+one `ClusterPolicyReport` for cluster-wide resources, see this [RFC](https://github.com/kubewarden/rfc/pull/13/files) for more info.
+
+Many more calls will be made to `PolicyServer` for the background checks, therefore we need to implement a cache
+for policy evaluations. The result of each policy evaluation will be stored in the cache, it will contain the `resourceVersion`
+of the policy and the resource to see if any of them have changed. Before evaluating a new resource it will check in the 
+cache if evaluation was previously done. 
+
+## Background Audit Scanner
+
+By default, background audit scanning happens every hour, and it can be configured via the `background-scan` flag.
+It will scan all resources in the cluster for all policies.
+When a background audit has finished processing all evaluations it will replace all `PolicyReports` and the `ClusterPolicyReport`
+with the new content. This will remove old entries with resources that are deleted and make sure the reports are up to date. 
 
-An audit checks consist of the following steps:
+Background Audit Scanner won't perform policy evaluation, it will make a request to `PolicyServer` for evaluating the resources
+instead.
+
+It should implement a caching mechanism to avoid overloading the Kubernetes api server. We can use client-go caching.
+
+A background audit checks consist of the following steps:
 1. Get information about which policies have audit checks enabled
 2. Get the resources configured in the policies with audit checks enabled
-3. Send the resources that a policy is configured to validate to it and get the validation results.
+3. Request the policy evaluation for all resources to the `PolicyServer`
 4. Aggregate all the policies validation results
-5. Publish a [PolicyReport](https://github.com/kubernetes-sigs/wg-policy-prototypes/tree/master/policy-report) with the aggregated results
+5. Publish a `PolicyReport` per namespace
+with the aggregated results and the `ClusterPolicyReport`.
 
 In step 2, after fetching the resources for the first time,  they will be stored in a
 cache and used again in futures audit checks until the *time to live* period expired.
 Triggering another request to the Kubernetes API.
 
-It necessary a configuration to define the *time to live* period used by the cache.
-
 ## ClusterAdmissionPolicy and AdmissionPolicy CRDs
 
-It's necessary a change in the policy CRDs to mark a policy to be run in the audit checks.
-The field would be called `audit` and its default value will be `true`.
+It's necessary a change in the policy CRDs to mark a policy to be run in the background audit checks.
+The field would be called `backgroundAudit` and its default value will be `true`.
```

Comment:
The same way that policies can be `mutating` or not, and `spec.mutating` is required, and listed in the policy metadata, I think we should do the same with `spec.backgroundAudit`. This allows for correct defaults on context-aware policies (they may not be audited? dunno).
What happens if a non-auditable policy gets `spec.backgroundAudit == True`?  Could that case happen?

Reply:
For example, if there is a policy that evaluates just UPDATE ops I think it should be ignored even if `spec.backgroundAudit == True`

Reply:
What happens if somebody misconfigures a policy that is checking `userInfo`, and for that policy they set `spec.backgroundAudit == True`?

Reply:
wrong data might be added to the report. We should specify in the docker that `userInfo` will be different in the docs, and suggest to set `spec.backgroundAudit == false` for policies that check `userInfo`

Reply:
Resolving, I consider it tackled with https://github.com/kubewarden/rfc/pull/10/commits/3172607122dadbd95a99e3433219816d2f3f9c59.